httpcore/_backends/anyio.py

```diff
@@ -135,8 +136,13 @@ async def open_tcp_stream(
     ) -> AsyncSocketStream:
         connect_timeout = timeout.get("connect")
         unicode_host = hostname.decode("utf-8")
+        exc_map = {
```

Comment:
Small nit: we could move the mapping to the class level. That way it's not created on each method call, and also could be shared between `open_tcp_stream()` and `open_uds_stream()` since the content is the same.
I see that the other backends have the same issue, so happy to defer this to a followup PR if we think it's worth doing.

Reply:
I like that this style makes it simple-as-cake to visually keep track of which exactly exceptions we map in each case, and I think moving it to a class variable (further away than from where it's actually used) wouldn't help with readability either. So overall, I'm thinking this may not really be worth a follow-up?